### DISCUSSION TITLE
Pod Disruption Budget is configurable

### DISCUSSION THREAD

**Discussion by niaomingjian** [emqx/emqx-operator#1092] [Q&A] [answered]
EMQX-Operator 2.2.18 introduced the Pod Disruption Budget (PDB) feature. From [the code](https://github.com/emqx/emqx-operator/compare/2.2.17...2.2.18#diff-a5bf658d0f423a13b8221c825dd388e5dd8d21214784424e4ac7e65214f85a6cR57), it is observed that MinAvailable is set to a fixed value of 1.
Is it possible to provide a configuration to disable PDB, or to set MinAvailable to 0?
If MinAvailable is fixed at 1, it may cause Kubernetes to fail during version upgrades.

**Comment by Rory-Z** [comment]
Hi, @niaomingjian I think you can just change the PDB which EMQX operator created, and set MinAvailable to 0.

**Reply by niaomingjian** [reply]
Yes, that's what I'm doing now. However, I think it might be more convenient if there were configurable features available.
When using Argo CD for deployments across multiple Kubernetes development environments, having this configuration option would be very helpful as it eliminates the need to manually update the configuration.

**Reply by Rory-Z** [reply]
I see, let me thinking it. what do you think about add `minAvailable / maxUnavailable` in `.spec.coreTemplate.spec` and `.spec.replicantTeamplte.spec` ?

**Comment by Rory-Z** [comment, marked as answer]
Hi, @niaomingjian EMQX operator 2.2.26 has been released, you can set `minAvailable / maxUnavailable` in `.spec.coreTemplate.spec` and `.spec.replicantTeamplte.spec`
https://github.com/emqx/emqx-operator/releases/tag/2.2.26
